cli-processor/src/main/java/gov/nist/secauto/metaschema/cli/processor/command/CommandExecutionException.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * SPDX-FileCopyrightText: none
+ * SPDX-License-Identifier: CC0-1.0
+ */
+
+package gov.nist.secauto.metaschema.cli.processor.command;
+
+import gov.nist.secauto.metaschema.cli.processor.ExitCode;
+import gov.nist.secauto.metaschema.cli.processor.ExitStatus;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * For use in commands to short-circut command execution.
+ */
+public class CommandExecutionException
+    extends Exception {
+  private final ExitCode exitCode;
+
+  /**
+   * the serial version UID.
+   */
+  private static final long serialVersionUID = 1L;
+
+  /**
+   * Constructs a new exception with the provided {@code code}, and no message or
+   * cause.
+   *
+   * @param code
+   *          the exit code associated with this error
+   */
+  public CommandExecutionException(@NonNull ExitCode code) {
+    this.exitCode = code;
+  }
+
+  /**
+   * Constructs a new exception with the provided {@code code}, {@code message},
+   * and no cause.
+   *
+   * @param code
+   *          the exit code associated with this error
+   * @param message
+   *          the exception message
+   */
+  public CommandExecutionException(@NonNull ExitCode code, String message) {
+    super(message);
+    this.exitCode = code;
+  }
+
+  /**
+   * Constructs a new exception with the no message and provided the {@code code}
```

Comment:
Very minor, non-blocking: grammar in docstring.

```suggestion
   * Constructs a new exception with no message and provided the {@code code}
```

Reply:
```suggestion
   * Constructs a new exception with no message and the provided {@code code}
```